waiter/src/waiter/client_tools.clj

```diff
@@ -574,7 +574,7 @@
   [waiter-url token & {:keys [hard-delete] :or {hard-delete true}}]
   (log/info "deleting token" token)
   (let [response (make-request waiter-url "/token"
-                               :headers {"host" token}
+                               :headers {"host" token, "if-match" (str (System/currentTimeMillis))}
```

Comment:
Delete should not require an `If-Match` so this should be removed.

Reply:
Deletes require the `if-match`|`if-unmodified-since` header since they can also participate in the concurrency race.

Reply:
Making them required would break everyone who is calling `POST` and `DELETE` that isn't the sync tool.  I'd like to see that our integration tests ensure that the headers are not required.

Reply:
It is only required when providing the hard-delete, changed the code to reflect this.